Readability.js

```diff
@@ -1316,6 +1316,61 @@ Readability.prototype = {
     return metadata;
   },
 
+  /**
+   * Find all <noscript> that located after <img> node, and contains exactly
+   * single <img> element. Once it found, this method will replace the previous <img>
+   * with <img> inside <noscript>, then finally remove the <noscript> tag. This is
+   * done because in some website (e.g. Medium), they use lazy load method like this.
+   *
+   * @param Element
+  **/
+  _unwrapNoscriptImages: function(doc) {
+    // First, find div which only contains single img element, then put it out.
+    var divs = doc.getElementsByTagName("div");
+    this._forEachNode(divs, function(div) {
+      if (div.children.length == 1 && div.children[0].tagName === "IMG") {
+        div.parentNode.replaceChild(div.children[0], div);
+      }
+    });
```

Comment:
I think this doesn't work correctly, because `getElementsByTagName` returns a live Node list, and you're then removing some of these nodes. On jsdom and other "proper" DOM implementations (unlike `JSDOMParser.js` in the readability repo), this would potentially skip some of the items.

I also think this is a little suboptimal from a performance perspective. In particular, I think there are normally fewer `img` elements than there are `div`s in the document, and so it would be more efficient to look for images, and then see if their parent is a div with no other children (and no inline siblings, see my other comment). Then you can also unify this loop with the next loop:

```js
var imgs = Array.from(doc.getElementsByTagName("img"));
this._forEachNode(imgs, function(img) {
  var src = img.getAttribute("src");
  var srcset = img.getAttribute("srcset");
  var dataSrc = img.getAttribute("data-src");
  var dataSrcset = img.getAttribute("data-srcset");

  if (!src && !srcset && !dataSrc && !dataSrcset) {
    img.parentNode.removeChild(img);
  } else if (img.parentNode.tagName == "DIV" && !img.nextElementSibling && !img.previousElementSibling) {
    var div = img.parentNode;
    // FIXME check for phrasing content that isn't empty/whitespace.
    div.parentNode.replaceChild(div, img);
  }
});
```